lib/after_commit_action.rb

```diff
@@ -17,6 +17,11 @@ module AfterCommitAction
   end
 
   def execute_after_commit(&block)
+    if ActiveRecord::Base.connection.open_transactions == 0 ||
+      ActiveRecord::Base.connection.add_transaction_record(self).nil?
+      return block.call
+    end
```

Comment:
There can be situation when there is a transaction, but current record is not added to transaction records. In that situation your code will not execute the block.
It is ok to add record multiple times to transaction, because rails executes #uniq on records array https://github.com/rails/rails/blob/5ba6999637361577539eafc63a39622ff21e345b/activerecord/lib/active_record/connection_adapters/abstract/transaction.rb#L84